pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -0,0 +1,198 @@
+limitations under the License.
+*/
+
+package mizar
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	v1 "k8s.io/arktos-ext/pkg/apis/arktosextensions/v1"
+	arktos "k8s.io/arktos-ext/pkg/generated/clientset/versioned"
+	arktoscheme "k8s.io/arktos-ext/pkg/generated/clientset/versioned/scheme"
+	arktosinformer "k8s.io/arktos-ext/pkg/generated/informers/externalversions/arktosextensions/v1"
+	arktosv1 "k8s.io/arktos-ext/pkg/generated/listers/arktosextensions/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+const (
+	mizarType = "mizar"
+)
+
+// MizarArktosNetworkController delivers grpc message to Mizar to update VPC with arktos network name
+type MizarArktosNetworkController struct {
+	netClientset    *arktos.Clientset
+	kubeClientset   *kubernetes.Clientset
+	netLister       arktosv1.NetworkLister
+	netListerSynced cache.InformerSynced
+	syncHandler     func(eventKey KeyWithEventType) error
+	queue           workqueue.RateLimitingInterface
+	recorder        record.EventRecorder
+	grpcHost        string
+}
+
+// NewMizarArktosNetworkController starts arktos network controller for mizar
+func NewMizarArktosNetworkController(netClientset *arktos.Clientset, kubeClientset *kubernetes.Clientset, informer arktosinformer.NetworkInformer, grpcHost string) *MizarArktosNetworkController {
+	utilruntime.Must(arktoscheme.AddToScheme(scheme.Scheme))
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(klog.Infof)
+	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: kubeClientset.CoreV1().EventsWithMultiTenancy(metav1.NamespaceAll, metav1.TenantAll)})
+
+	c := &MizarArktosNetworkController{
+		netClientset:    netClientset,
+		kubeClientset:   kubeClientset,
+		netLister:       informer.Lister(),
+		netListerSynced: informer.Informer().HasSynced,
+		queue:           workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		recorder:        eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "mizar-arktos-network-controller"}),
+		grpcHost:        grpcHost,
+	}
+
+	informer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: c.createNetwork,
+	})
+
+	c.netLister = informer.Lister()
+	c.netListerSynced = informer.Informer().HasSynced
+	c.syncHandler = c.syncNetwork
+
+	return c
+}
+
+// Run update from mizar cluster
+func (c *MizarArktosNetworkController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+	klog.Info("starting network controller")
```

Comment:
log message: "start mizar arktos controller", to differ from the arktos controller.